src/Configuration/NodeDefinition/NodeDefinitionInterface.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractorConfig\Configuration\NodeDefinition;
+
+use Symfony\Component\Config\Definition\Builder\NodeDefinition;
+
+interface NodeDefinitionInterface
```

Comment:
Not sure if there is a point to make an interface. Do you typehint it somewhere?

Reply:
Tohle není tak úplně node definition interface, ne? NodeDefinitionInterface je to co by implementoval NodeDefinition, ne? Tohle je spíš NodeDefinitionBuilderInterface nebo NodeDefinitionCreatorInterface - je potenciál, že se to použije ještě někde, tak by se to mělo jmenovat správně/pochopitelně k tomu co to dělá. 

A když už to děláš takhle a nic jiného to nedělá, tak by to klidně mohlo dědit přímo od ArrayNodeDefinition a celou tu práci s vytvořením těch podpoložek to udělat ve svém konstruktoru (resp. v nějaké metodě init nebo klidně můžeš nechat create, která se spustí v konstruktoru). Když se podíváš jak funguje třeba `scalarNode()`, tak jediné co to udělá je new ScalarNode() a append() (viz \Symfony\Component\Config\Definition\Builder\NodeBuilder::node). Takže `SshNode extends ArrayNodeDefinition` a tenhle interface můžeš úplně zahodit a typehintovat v těch classách buď přímo SshNode nebo obecně nějaký NodeDefinition.